src/engine/controls/cuecontrol.cpp

```diff
@@ -1173,6 +1208,44 @@ void CueControl::hotcueActivatePreview(HotcueControl* pControl, double value) {
     setHotcueFocusIndex(pControl->getHotcueIndex());
 }
 
+void CueControl::loopCueReloop(HotcueControl* pControl, const CuePointer& pCue) {
+    if (pCue->getType() != mixxx::CueType::Loop) {
+        return;
+    }
+
+    if (m_pCurrentSavedLoopControl != pControl) {
+        setCurrentSavedLoopControlAndActivate(pControl);
+    } else {
+        bool loopActive = pControl->getStatus() ==
+                HotcueControl::Status::Active;
+        Cue::StartAndEndPositions pos = pCue->getStartAndEndPosition();
+        setLoop(pos.startPosition, pos.endPosition, !loopActive);
+    }
+}
+
+void CueControl::loopCueGoToAndLoopOrToggle(HotcueControl* pControl, const CuePointer& pCue) {
+    if (pCue->getType() != mixxx::CueType::Loop) {
+        return;
+    }
+
+    if (m_pCurrentSavedLoopControl != pControl) {
+        setCurrentSavedLoopControl(pControl);
+    }
+    Cue::StartAndEndPositions pos = pCue->getStartAndEndPosition();
+    mixxx::audio::FramePos currPos = getQuantizedCurrentPosition();
+    if (currPos < pos.startPosition || currPos >= pos.endPosition) {
+        // if we're ahead or behind loop: jump to and activate
+        seekAbs(pCue->getPosition());
+        m_pLoopEnabled->set(1.0);
+        pControl->setStatus(HotcueControl::Status::Active);
+    } else {
+        // we're inside loop range: toggle on/off
```

Comment:
Do we want this case here? As @nPrevail has described he likes to use the loop-cue like a normal cue just with a loop boundary for cuing in. For disabling an enabling we have the secondary function and the reloop button. 
I think I would prefer to remove the OrToggle part.

Reply:
Exactly. I will update this PR accordingly.

Reply:
The secondary function may not be available, see https://github.com/mixxxdj/mixxx/pull/4816#issuecomment-1963097951